pymodm/fields.py

```diff
@@ -1180,15 +1199,23 @@ def contribute_to_class(self, cls, name):
                                                     self._on_delete)
 
     def to_python(self, value):
+        doc = value
         if isinstance(value, dict):
             # Try to convert the value into our document type.
-            return self.related_model.from_document(value)
-        elif isinstance(value, self.related_model):
-            return value
+            try:
+                doc = self.related_model.from_document(value)
+            except Exception:
+                pass
+
+        if isinstance(doc, self.related_model):
+            return doc
         elif self.model._mongometa._auto_dereference:
             # Attempt to dereference the value as an id.
             dereference_id = _import('pymodm.dereference.dereference_id')
             return dereference_id(self.related_model, value)
+
+        # TODO: should we convert value to mongo here?
+        # return self.related_model._mongometa.pk.to_mongo(value)
```

Comment:
After this change this test https://github.com/mongodb/pymodm/blob/ce603f5079fb602eeb2421e1bca767f4cb2bd8ac/test/test_related_fields.py#L60
start to fail cause it try to convert 1234 into ObjectId. As 1234 is not a valid value is that ok that full_clean does not raise ValidationError?